src/test/kotlin/com/example/realworldkotlinspringbootjdbc/infra/CommentRepositoryImplTest.kt

```diff
@@ -244,4 +244,161 @@ class CommentRepositoryImplTest {
+            ignoreCols = ["id", "created_at", "updated_at"],
+            orderBy = ["id"]
+        )
+        fun `準正常系-「articles テーブルに slug に該当する記事が存在」「comments テーブルに commentId に該当するコメントが存在しない」場合、削除失敗し、戻り値が CommentNotFoundByCommentId`() {
+            /**
+             * given:
+             */
+            val commentRepository = CommentRepositoryImpl(namedParameterJdbcTemplate)
+
+            /**
+             * when:
+             * - slug: articles テーブルに存在する
+             * - commentId: article_comments テーブルに存在しない
+             * - currentUserId: このテストで値に意味がない
+             */
+            val actual = commentRepository.delete(
+                slug = Slug.newWithoutValidation("rust-vs-scala-vs-kotlin"),
+                commentId = CommentId.newWithoutValidation(100), // 存在しない CommentId
+                currentUserId = UserId(3)
+            )
+
+            /**
+             * then:
+             */
+            val expected = CommentRepository.DeleteError.NotFoundCommentByCommentId(
+                slug = Slug.newWithoutValidation("rust-vs-scala-vs-kotlin"),
+                commentId = CommentId.newWithoutValidation(100),
+                currentUserId = UserId(3)
+            )
+            when (actual) {
+                is Left -> assertThat(actual.value).isEqualTo(expected)
+                is Right -> assert(false)
+            }
+        }
+
+        @Test
+        @DataSet("datasets/yml/given/articles.yml")
+        @ExpectedDataSet(
+            value = ["datasets/yml/given/articles.yml"],
+            ignoreCols = ["id", "created_at", "updated_at"],
+            orderBy = ["id"]
+        )
+        fun `準正常系-「articles テーブルに slug に該当する記事が存在」「comments テーブルに commentId に該当するコメントが存在」「コメントの authorId が currentUserId と異なる」場合、削除失敗し、戻り値が NotAuthorizedDeleteComment`() {
+            /**
+             * given:
+             */
+            val commentRepository = CommentRepositoryImpl(namedParameterJdbcTemplate)
+
+            /**
+             * when:
+             * - slug: articles テーブルに存在する
+             * - commentId: article_comments テーブルに存在する
+             * - currentUserId: commentId に該当する article_comments テーブルのレコードの authorId と一致しない
+             */
+            val actual = commentRepository.delete(
+                slug = Slug.newWithoutValidation("rust-vs-scala-vs-kotlin"),
+                commentId = CommentId.newWithoutValidation(1),
+                currentUserId = UserId(100) // コメントの authorId と 一致しない
+            )
+
+            /**
+             * then:
+             */
+            val expected = CommentRepository.DeleteError.NotAuthorizedDeleteComment(
+                slug = Slug.newWithoutValidation("rust-vs-scala-vs-kotlin"),
+                commentId = CommentId.newWithoutValidation(1),
+                currentUserId = UserId(100)
+            )
+            when (actual) {
+                is Left -> assertThat(actual.value).isEqualTo(expected)
+                is Right -> assert(false)
+            }
```

Comment:
[imo]

以下のようにして、直接比較するのはどうでしょう？

```suggestion
            val expected = CommentRepository.DeleteError.NotAuthorizedDeleteComment(
                slug = Slug.newWithoutValidation("rust-vs-scala-vs-kotlin"),
                commentId = CommentId.newWithoutValidation(1),
                currentUserId = UserId(100)
            ).left()
            assertThat(actual).isEqualTo(expected)
```

Reply:
こちら修正しました。

https://github.com/sunakan/realworld-kotlin-springboot-jdbc/pull/109/commits/6806bcbc68ff229631303bdd391a07c79b62deab